include/pthread.h

```diff
@@ -197,7 +197,7 @@
 #define pthread_setname_np(thread, name) \
   prctl((int)PR_SET_NAME_EXT, (char*)name, (int)thread)
 
-#define pthread_getname_np(thread, name) \
+#define pthread_getname_np(thread, name, len) \
```

Comment:
Shouldn't we pass `len` argument to `prctl` call?

Reply:
And also process this argument when copy task name to check length

Reply:
It's hard to do that without introduce a new structure since the prototype of prctl is fixed to three argument. The assumption is that the buffer passed by caller should be at least CONFIG_TASK_NAME_SIZE + 1.

Reply:
I searched the prototype and it was variable parameter count prototype. Or maybe I visited a wrong file.

Reply:
Yes, you are right, but PR_GET_NAME defined by Linux in the same way:
https://man7.org/linux/man-pages/man2/prctl.2.html
```
 PR_GET_NAME (since Linux 2.6.11)
              Return the name of the calling thread, in the buffer
              pointed to by (char *) arg2.  The buffer should allow
              space for up to 16 bytes; the returned string will be
              null-terminated.
```
So, it's better to keep the prototype as it.

Reply:
why don't you make it a function which perform the size check before calling prctl?
i think it's how it's implemented in linux.

Reply:
No, please see my above reply.

Reply:
Good idea, done.